src/DataValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace League\ISO3166;
+
+class DataValidator
+{
+    use KeyValidators;
+
+    /**
+     * @param array $data
+     *
+     * @return array
+     */
+    public function validate(array $data)
+    {
+        foreach ($data as $entry) {
+            $this->assertEntryHasRequiredKeys($entry);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $entry
+     *
+     * @throws \DomainException if given data entry does not have all the required keys.
+     */
+    private function assertEntryHasRequiredKeys(array $entry)
```

Comment:
What exactly would be the use-case for validating just one entry? And even if you wanted to validate just one entry, you could simply wrap it in an array and pass it to validate

I would prefer to keep the public API as minimal as possible. This allows for more room to introduce changes in the validation process.